common/monitoring/monitoring.go

```diff
@@ -154,33 +155,30 @@ func handleFunc(endpointName string) {
 //
 // If we attempt send more messages than the size of the buffer, these overflowing messages will be ignored and warning will be logged.
 func Run(port uint16, endpointName string) error {
-	if IsRunning() {
+	srv := &http.Server{Addr: fmt.Sprintf(":%d", port)}
+	// only one Run should initialize and serve
+	if !server.CompareAndSwap(nil, srv) {
```

Comment:
Sorry, I get very much confused when trying to read this and remembering which server/srv is which. Could you perhaps call them `serverAtomicPtr` and just `server`, or anything else which conveys the difference?